modules/miner/miner.go

```diff
@@ -28,12 +34,13 @@ type Miner struct {
 
 	subscribers []chan struct{}
 
-	mu sync.RWMutex
+	mu sync.Mutex
```

Comment:
I thought it might be faster. Benchmarking seemed to indicate that it was a tiny bit faster, so I just left it. I think there's overhead when you start using readlocks instead of plainlocks.